addons/mail/static/src/js/client_action.js

```diff
@@ -103,6 +103,74 @@ var PartnerInviteDialog = Dialog.extend({
     },
 });
 
+/**
+ * Widget : Moderator reject message dialog
+ *
+ * Popup containing message title and reject message body.
+ */
+var ModeratorRejectMessageDialog = Dialog.extend({
+    template: "mail.ModeratorRejectMessageDialog",
+    /**
+     * @override
+     */
+    init: function (parent, messageIDs) {
+        this.messageIDs = messageIDs;
+        this._super(parent, {
+            title: _t('Send Mail to Author'),
+            size: "medium",
+            buttons: [{
+                text: _t("Send"),
+                close: true,
+                classes: "btn-primary",
+                click: _.bind(this._onSendClicked, this),
+            }],
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @return {string[]} list of emails
+     */
+    _getEmailTo: function () {
+        var messages = _.map(this.messageIDs, function (messageID) {
+            return chat_manager.get_message(messageID);
+        });
+        return _.pluck(messages, 'email_from').join(',');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+  
+    /**
+     * Bind handler for send notification on rejection of message by moderator
+     *
+     * @private
+     */
+    _onSendClicked: function () {
+        var self = this;
+        var message = this.$('#reject_message').val();
+        var title = this.$('#message_title').val();
+        if (message && title) {
+            this._rpc({
+                model: 'mail.message',
+                method: 'create_notification_email',
```

Comment:
Just directly create a mail.mail with state = outgoing. create_notification_email is an unnecessary wrapper.

Reply:
done